NativeApp/src/components/Home/container.js

```diff
@@ -18,20 +21,112 @@ export default Container => class extends Component {
       super(props);
       this.state = {
         takenHolidays: {},
+        showModal: false,
+        user: null,
+        booking: {
+          startDate: '',
+          endDate: '',
+          halfDay: false,
+        },
       };
     }
 
     componentDidMount() {
       getTakenHolidays()
         .then(data => this.setState({ takenHolidays: this.formatDate(data) }));
+
+      userProfile()
+        .then(user => this.setState({ user }));
+    }
+
+    onDayPress = (day) => {
+      if (day) {
+        this.setState({
+          showModal: true,
+          booking: {
+            startDate: day.dateString,
+            endDate: day.dateString,
+          },
+        });
+      }
+    }
+
+    submitRequest = () => {
+      const { booking, user } = this.state;
+      const request = {
+        dates: [
+          {
+            endDate: booking.endDate,
+            halfDay: false,
+            startDate: booking.startDate,
+          },
+        ],
+        employeeId: user.employeeId,
+      };
+
+      requestHolidays(request)
+        .then(() => {
+          getTakenHolidays()
+            .then(data => this.setState({ takenHolidays: this.formatDate(data) }));
+          this.closeModal();
+        })
+        .catch(e => Alert.alert(
+          'Could not request holidays',
+          e.message,
+        ));
+    }
+
+    changeStartDate = (date) => {
+      const formatDate = moment(date).format('YYYY-MM-DD');
+      this.setState(prevState => ({
+        booking: {
+          ...prevState.booking,
+          startDate: formatDate,
+        },
+      }));
+    }
+
+    changeEndDate = (endDate) => {
+      const formatEndDate = moment(endDate).format('YYYY-MM-DD');
+      this.setState(prevState => ({
+        booking: {
+          ...prevState.booking,
+          endDate: formatEndDate,
+        },
+      }));
+    }
+
+    closeModal = () => {
+      const { showModal } = this.state;
+      this.setState({ showModal: !showModal });
```

Comment:
should we change `!showModal` to `false` here?
I appreciate that if the modal is open `showModal` should be `true` - but I'm just wary incase future code messes that state setting up.

Reply:
Good point! Safer to set it to false.